meson.build

```diff
@@ -72,7 +72,7 @@ endif
 if host_machine.system() == 'windows'
   r2birth = run_command('cmd', '/c', 'echo %date%__%time%')
 else
-  r2birth = run_command('date', '+%Y-%m-%d__%H:%M:%S')
+  r2birth = run_command('sh', '-c', 'SOURCE_DATE_EPOCH="${SOURCE_DATE_EPOCH:-`date +%s`}"; FORMAT="+%Y-%m-%d__%H:%M:%S"; date -u -d @"$SOURCE_DATE_EPOCH" "$FORMAT" 2>/dev/null || date -u -r "$SOURCE_DATE_EPOCH" "$FORMAT" 2>/dev/null || date -u "$FORMAT"')
```

Comment:
that line requires bash and its pretty unreadable, maybe better to move the logic into sys/date.sh and make the code cleaner here.

Reply:
I thought, it should work with any POSIX-compliant shell. Or did I miss a bash-ism? A quick with Debian's `dash` looked good. A `sys/date.sh` would probably need a shell, too.
For readability, I thought about adding linebreaks, but not sure what escaping is needed in meson.

Reply:
having it in a separate script would also make it easier to use it in the alternative configure/make build sys, i guess

Reply:
I changed it into a multi-line string for now. I'm off to bed now. Feel free to push the move to `date.sh` to this PR.

Reply:
Oh, i was mistaken, the ${:-} is posix and works on mksh,dash,zsh .. so definively not a bashism. its way more readable this way :) thanks for the improvement